Cargo.toml

```diff
@@ -22,6 +23,11 @@ string-interner = { version = "0.14.0", default-features = false, features = ["s
 take_mut = "0.2.2"
 try-from-discrim = "1.0.0"
 warnings-gen = { path = "warnings-gen" }
+nix = "0.28.0"
+clap-num = "1.1.1"
+thiserror = "1.0.58"
+tracing = "0.1.40"
+tracing-subscriber = "0.3.18"
```

Comment:
We're already pulling in `codespan-reporting` as an error reporting crate, I'd like all of RGBDS to be consistent about it.

(Preferably, all programs would pull in a common module, but I'm fine with the PR duplicating work for the sake of simplicity; I can perform the deduplication refactoring myself afterwards.)

Reply:
Interesting. This change was Evie's pick, and she said she thinks codespans should not be used for error reporting, but I understand the need for consistency.

Reply:
Yeah, I agree it's arguably abuse of the crate... but I also think consistency, being user-facing, trumps that.

Reply:
For what it's worth, I have pushed [a commit that switches the diagnostics backend to `ariadne`](https://github.com/ISSOtm/rsgbds/commit/891b78e8562456de90524c79b59631d7d986ce6a). It should have a much more practical API than `codespan_reporting` for rgbfix.